MAPIStoreFunctions.cpp

```diff
@@ -831,7 +832,19 @@ _Check_return_ HRESULT OpenPublicMessageStore(
 
 	if (lpPublicMDBNonAdmin && StoreSupportsManageStore(lpPublicMDBNonAdmin))
 	{
-		EC_MAPI(HrGetOneProp(lpPublicMDBNonAdmin, CHANGE_PROP_TYPE(PR_HIERARCHY_SERVER, PT_STRING8), &lpServerName));
+		if (szServerName != nullptr && strlen(szServerName) > 0)

```

Comment:
auto server = szServerName;
if (server.Empty())
{
do getprops, copy lpszA to server, mapifreebuffer
}

if (!server.empty())
{
  pickup from BuildServerDN code here
...